src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php

```diff
@@ -85,12 +86,54 @@ protected function setup_template_vars() {
 		$template_vars = parent::setup_template_vars();
 
 		// Here update, add and remove from the default template vars.
-		$template_vars['view_more_link']   = tribe_get_events_link();
-		$template_vars['widget_title']     = $this->context->get( 'widget_title' );
-		$template_vars['show_latest_past'] = false;
+		$template_vars['view_more_link']     = tribe_get_events_link();
+		$template_vars['widget_title']       = $this->context->get( 'widget_title' );
+		$template_vars['no_upcoming_events'] = $this->context->get( 'no_upcoming_events' );
+		$template_vars['show_latest_past']   = false;
 		// Display is modified with filters in Pro.
 		$template_vars['display'] = [];
 
 		return $template_vars;
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function setup_messages( array $events ) {
+		if ( ! empty( $events )	) {
+			return;
+		}
+
+		$keyword = $this->context->get( 'keyword', false );
+		$this->messages->insert(
+			Messages::TYPE_NOTICE,
+			Messages::for_key( 'no_upcoming_events', trim( $keyword ) )
+		);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setup_the_loop( array $args = [] ) {
```

Comment:
I added this method here as tests were failing due to an undefined index. 

This is the coding causing it: ( I think because we do not register rewrite rules for the view because it is a widget )
// Set the $_SERVER['REQUEST_URI'] as many WordPress functions rely on it to correctly work.
$_SERVER['REQUEST_URI'] = $this->get_request_uri();

Since the widget is not called by a url, I removed it. 

Open to alternatives. 

This is the error I get when that line is included:
Undefined index: path
 wp-includes/canonical.php:576
 wp-content/plugins/the-events-calendar/common/src/Tribe/Rewrite.php:567
 wp-content/plugins/the-events-calendar/common/src/Tribe/Rewrite.php:1098
 wp-content/plugins/the-events-calendar/src/Tribe/Views/V2/View.php:1540
 wp-content/plugins/the-events-calendar/src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php:136
 wp-content/plugins/the-events-calendar/src/Tribe/Views/V2/View.php:603
 wp-content/plugins/the-events-calendar/tests/views_widgets/Tribe/Events/Views/V2/Widgets/Widget_ListTest.php:143

Reply:
Since this is a widget, ideally the `setup_the_loop` method should be overridden, as you're doing here, to make it do nothing.
What does happen if you make it do nothing?

Reply:
I went ahead and remove everything from the method, but the lines to set the repository args. The widget works and all the global overwrites are gone.